File: src/Entity/ContratMaintenance.php
<?php

namespace App\Entity;

use App\Repository\ContratMaintenanceRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: ContratMaintenanceRepository::class)]
class ContratMaintenance
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    private ?string $numContrat = null;

    #[ORM\Column(nullable: true)]
    private ?\DateTime $dateSignature = null;

    #[ORM\Column(nullable: true)]
    private ?\DateTime $dateEcheance = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNumContrat(): ?string
    {
        return $this->numContrat;
    }

    public function setNumContrat(string $numContrat): static
    {
        $this->numContrat = $numContrat;

        return $this;
    }

    public function getDateSignature(): ?\DateTime
    {
        return $this->dateSignature;
    }

    public function setDateSignature(?\DateTime $dateSignature): static
    {
        $this->dateSignature = $dateSignature;

        return $this;
    }

    public function getDateEcheance(): ?\DateTime
    {
        return $this->dateEcheance;
    }

    public function setDateEcheance(?\DateTime $dateEcheance): static
    {
        $this->dateEcheance = $dateEcheance;

        return $this;
    }
}

File: ClientLourd/src/main/java/com/cashcash/models/ContratMaintenance.java
package com.cashcash.models;

import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlRootElement;
import java.time.LocalDate;

/**
 * Classe représentant un contrat de maintenance d'un matériel.
 * Permet de déterminer si un matériel est "sous contrat" ou non.
 */
@XmlRootElement(name = "ContratMaintenance")
public class ContratMaintenance {

    private int id;
    private String numContrat;
    private LocalDate dateSignature;
    private LocalDate dateEcheance;
    private String typeContrat;

    public ContratMaintenance() {}

    public ContratMaintenance(int id, String numContrat, LocalDate dateSignature, LocalDate dateEcheance, String typeContrat) {
        this.id = id;
        this.numContrat = numContrat;
        this.dateSignature = dateSignature;
        this.dateEcheance = dateEcheance;
        this.typeContrat = typeContrat;
    }

    @XmlElement(name = "Id")
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    @XmlElement(name = "NumContrat")
    public String getNumContrat() {
        return numContrat;
    }

    public void setNumContrat(String numContrat) {
        this.numContrat = numContrat;
    }

    @XmlElement(name = "DateSignature")
    public LocalDate getDateSignature() {
        return dateSignature;
    }

    public void setDateSignature(LocalDate dateSignature) {
        this.dateSignature = dateSignature;
    }

    @XmlElement(name = "DateEcheance")
    public LocalDate getDateEcheance() {
        return dateEcheance;
    }

    public void setDateEcheance(LocalDate dateEcheance) {
        this.dateEcheance = dateEcheance;
    }

    @XmlElement(name = "TypeContrat")
    public String getTypeContrat() {
        return typeContrat;
    }

    public void setTypeContrat(String typeContrat) {
        this.typeContrat = typeContrat;
    }

    /**
     * Vérifie si le contrat est actif (entre la date de signature et d'échéance).
     * @return true si le contrat est actif, false sinon
     */
    public boolean isActive() {
        LocalDate today = LocalDate.now();
        return !today.isBefore(dateSignature) && !today.isAfter(dateEcheance);
    }

    /**
     * Vérifie si le contrat est expiré.
     * @return true si le contrat est expiré, false sinon
     */
    public boolean isExpired() {
        return LocalDate.now().isAfter(dateEcheance);
    }
}
